app/report-data/get-claim-level-report-data.js

```diff
@@ -0,0 +1,54 @@
+const { bps, cs, fdmr } = require('../constants/source-systems')
+const db = require('../data')
+
+const getClaimLevelReportData = async (startDate, endDate) => {
+  let startDateEndDate = ''
+  if (startDate && endDate) {
+    startDateEndDate = `WHERE "lastUpdated" BETWEEN '${startDate}' AND '${endDate}'`
+  }
+  return db.sequelize.query(`
+  WITH "rankedData" AS (
+    SELECT
+      *,
+      ROW_NUMBER() OVER (
+        PARTITION BY
+          "sourceSystem",
+          frn,
+          CASE 
+              WHEN "sourceSystem" = :BPS THEN NULL
+              WHEN "sourceSystem" = :FDMR THEN NULL
+              WHEN "sourceSystem" = :CS THEN "claimNumber"
+              ELSE "agreementNumber"
+          END,
+          CASE
+            WHEN "sourceSystem" = :FDMR THEN NULL
+            WHEN "sourceSystem" = :CS THEN NULL
+            ELSE "marketingYear"
+          END
+        ORDER BY
+          "paymentRequestNumber" DESC,
+          "lastUpdated" DESC
+      ) AS row_num
+    FROM
+      "reportData"
+      ${startDateEndDate}
```

Comment:
my ocd would prefer the ${startDateEndDate} to be aligned with FROM, but that's just ocd. up to you if you can be bothered doing this!